docs/content/library/AsyncArrow.fsx

```diff
@@ -0,0 +1,339 @@
+(**
+
+# F# Async: AsyncArrow
```

Comment:
No particular reason, I'll change it to `AsyncFunc`. The reason I went with arrow is because it was an existing term and sounded at least somewhat intuitive. I initially considered `AsyncFun`, but clearly that is a bit much.